src/metabase/server/middleware/session.clj

```diff
@@ -356,10 +356,23 @@
   ;; The amount is nillable.
   (deferred-tru "Time before inactive users are logged out. By default, sessions last indefinitely.")
   :type       :json
-  :default    nil)
+  :default    nil
+  :setter     (fn [new-value]
+                (when (some? new-value)
+                  ;; Assert the duration is less than 100 years.
+                  ;; Otherwise, we could fail to format the expires date if the year
+                  ;; has more than 4 digits (#25253)
+                  (let [{:keys [unit amount]} new-value
+                        days (.toDays (case unit
+                                        "seconds" (t/seconds amount)
+                                        "minutes" (t/minutes amount)
+                                        "hours"  (t/hours amount)))
+                        years (/ days 365.25)]
+                    (assert (< years 100) "Session timeout must be less than 100 years")))
+                (setting/set-value-of-type! :json :session-timeout new-value)))
```

Comment:
This doesn't really cover the case of someone setting this to something insane if they set it via env var, or handle cases where they set it to something insane before this check was in place. Does it make sense to have some logic in the `:getter` as well to make sure it returns a sane value?

Reply:
>  if they set it via env var

I don't think we have any validation for settings set by env vars in general. Perhaps we should add a general validation function property to `defsetting`, so env vars can be validated with the same logic as the setter?

> or handle cases where they set it to something insane before this check was in place.

If they set it to something insanely before this check was in place, then they'd have run into the error this PR fixes and their Metabase instance would be producing errors everywhere. I don't think it's in scope to fix that.

Reply:
@camsaul I've added validation via on the getter fn in this commit for env vars

Let me know if you think this is sufficient
https://github.com/metabase/metabase/pull/27347/commits/dc594c619096059a3c3288f3a402337e1c4305a4